### PR TITLE
Add support for detailed PSQL user permissions

### PR DESCRIPTION
**Pull request by tun0**

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -22,5 +22,22 @@ locals {
     # default host in the resulting user object as well
     for user in var.users : "${user.name}@${user.host == null ? "%" : user.host}" => defaults(user, { host = "%" })
   }
-  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => user }
+  postgres_admins  = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => user if user.admin}
+  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => defaults(user, { db_ro = "", db_rw = ""}) if user.admin == false}
```
Comment (EvyBongers) [inline locals.tf]: Are `db_ro` and `db_rw` not supposed to be lists?
```suggestion
  postgres_users = local.db_engine != "POSTGRES" ? {} : { for user in var.users : user.name => defaults(user, { db_ro = [], db_rw = []}) if user.admin == false}
```

Comment (tun0) [reply locals.tf]: You'd think so, yes. But, no, `terraform` doesn't accept that.
```
│ Invalid value for "defaults" parameter: .db_ro: invalid default value for string: string required.
```

**Comment by tun0** [comment]
```terraform
  users              = [
    {name = "app", admin = false, db_rw = ["homeowner-leads"]},
    {name = "dev", admin = false, db_ro = ["homeowner-leads"]},
  ]
```

yields:
```terraform

  # module.db.postgresql_grant.postgres_readers["homeowner-leads/dev"] will be created
  + resource "postgresql_grant" "postgres_readers" {
      + database          = "homeowner-leads"
      + id                = (known after apply)
      + object_type       = "table"
      + privileges        = [
          + "SELECT",
        ]
      + role              = "dev"
      + schema            = "public"
      + with_grant_option = false
    }

  # module.db.postgresql_grant.postgres_writers["homeowner-leads/app"] will be created
  + resource "postgresql_grant" "postgres_writers" {
      + database          = "homeowner-leads"
      + id                = (known after apply)
      + object_type       = "table"
      + privileges        = [
          + "INSERT",
          + "SELECT",
          + "UPDATE",
        ]
      + role              = "app"
      + schema            = "public"
      + with_grant_option = false
    }

  # module.db.postgresql_role.postgres_users["app"] will be created
  + resource "postgresql_role" "postgres_users" {
      + bypass_row_level_security = false
      + connection_limit          = -1
      + create_database           = false
      + create_role               = false
      + encrypted_password        = true
      + id                        = (known after apply)
      + inherit                   = true
      + login                     = true
      + name                      = "app"
      + password                  = (sensitive value)
      + replication               = false
      + skip_drop_role            = false
      + skip_reassign_owned       = false
      + superuser                 = false
      + valid_until               = "infinity"
    }

  # module.db.postgresql_role.postgres_users["dev"] will be created
  + resource "postgresql_role" "postgres_users" {
      + bypass_row_level_security = false
      + connection_limit          = -1
      + create_database           = false
      + create_role               = false
      + encrypted_password        = true
      + id                        = (known after apply)
      + inherit                   = true
      + login                     = true
      + name                      = "dev"
      + password                  = (sensitive value)
      + replication               = false
      + skip_drop_role            = false
      + skip_reassign_owned       = false
      + superuser                 = false
      + valid_until               = "infinity"
    }

  # module.db.random_password.sql_user["app"] will be created
  + resource "random_password" "sql_user" {
      + id          = (known after apply)
      + length      = 48
      + lower       = true
      + min_lower   = 0
      + min_numeric = 0
      + min_special = 0
      + min_upper   = 0
      + number      = true
      + result      = (sensitive value)
      + special     = false
      + upper       = true
    }

  # module.db.random_password.sql_user["dev"] will be created
  + resource "random_password" "sql_user" {
      + id          = (known after apply)
      + length      = 48
      + lower       = true
      + min_lower   = 0
      + min_numeric = 0
      + min_special = 0
      + min_upper   = 0
      + number      = true
      + result      = (sensitive value)
      + special     = false
      + upper       = true
    }
```

File: locals.tf (hunk 1 above)
Comment (EvyBongers) [reply locals.tf]: Ah, right. That's where that problem is. Maybe `list(string)`?

Comment (tun0) [reply locals.tf]: That'd make those parameters required and indeed avoid the whole `null` issue. But at the cost at a more verbose configuration. I *might* go that route after splitting off the `admin` stuff from the *regular* users.

**Comment by tun0** [comment]
To be moved to separate module.

**Pull request closed without merging**